70_FLIPFluids/gpuRadixSort.h

```diff
@@ -0,0 +1,470 @@
+            }
+
+			cmdbuf->bindComputePipeline(m_radixSortPipeline.get());
+			const IGPUDescriptorSet* sortSet = m_radixSortDs[i % 2].get();
+		    cmdbuf->bindDescriptorSets(nbl::asset::EPBP_COMPUTE, m_radixSortPipeline->getLayout(), 1, 1, &sortSet);
+		    cmdbuf->dispatch(numWorkgroups, 1, 1);
+
+			if (i < numIterations - 1) {
+				SMemoryBarrier memBarrier;
+				memBarrier.srcStageMask = PIPELINE_STAGE_FLAGS::COMPUTE_SHADER_BIT;
+				memBarrier.srcAccessMask = ACCESS_FLAGS::SHADER_WRITE_BITS;
+				memBarrier.dstStageMask = PIPELINE_STAGE_FLAGS::COMPUTE_SHADER_BIT;
+				memBarrier.dstAccessMask = ACCESS_FLAGS::SHADER_READ_BITS;
+				cmdbuf->pipelineBarrier(E_DEPENDENCY_FLAGS::EDF_NONE, {.memBarriers = {&memBarrier, 1}});
+            }
+        }
+    }
+
+private:
+    bool updateBuffers(int numElements, int numWorkgroups, int dataTypeSize)
+    {
+		bool updated = false;
+        if (!tempDataBuffer || tempDataBuffer->getSize() != numElements * dataTypeSize)
+        {
+            video::IGPUBuffer::SCreationParams params = {};
+		    params.size = numElements * dataTypeSize;
+		    params.usage = IGPUBuffer::EUF_STORAGE_BUFFER_BIT;
+            tempDataBuffer = m_device->createBuffer(std::move(params));
+
+		    video::IDeviceMemoryBacked::SDeviceMemoryRequirements reqs = tempDataBuffer->getMemoryReqs();
+		    reqs.memoryTypeBits &= m_device->getPhysicalDevice()->getDeviceLocalMemoryTypeBits();
+
+		    auto bufMem = m_device->allocate(reqs, tempDataBuffer.get());
+			updated = true;
+        }
+		uint32_t bufSize = sizeof(uint32_t) * numIterations * NUM_SORT_BINS;
+		if (!globalHistogramsBuffer || globalHistogramsBuffer->getSize() != bufSize)
+		{
+			video::IGPUBuffer::SCreationParams params = {};
+		    params.size = bufSize;
+		    params.usage = IGPUBuffer::EUF_STORAGE_BUFFER_BIT | IGPUBuffer::EUF_TRANSFER_DST_BIT;
+            globalHistogramsBuffer = m_device->createBuffer(std::move(params));
+
+		    video::IDeviceMemoryBacked::SDeviceMemoryRequirements reqs = globalHistogramsBuffer->getMemoryReqs();
+		    reqs.memoryTypeBits &= m_device->getPhysicalDevice()->getDeviceLocalMemoryTypeBits();
+
+		    auto bufMem = m_device->allocate(reqs, globalHistogramsBuffer.get());
+			updated = true;
+		}
+		bufSize = sizeof(uint32_t) * (numWorkgroups * NUM_SORT_BINS + 1);
+		if (!partitionHistogramBuffer || partitionHistogramBuffer->getSize() != bufSize)
+		{
+			video::IGPUBuffer::SCreationParams params = {};
+		    params.size = bufSize;
+		    params.usage = IGPUBuffer::EUF_STORAGE_BUFFER_BIT;
+            partitionHistogramBuffer = m_device->createBuffer(std::move(params));
+
+		    video::IDeviceMemoryBacked::SDeviceMemoryRequirements reqs = partitionHistogramBuffer->getMemoryReqs();
+		    reqs.memoryTypeBits &= m_device->getPhysicalDevice()->getDeviceLocalMemoryTypeBits();
+
+		    auto bufMem = m_device->allocate(reqs, partitionHistogramBuffer.get());
+			updated = true;
+		}
+
+		return updated;
```

Comment:
first thing, oh the horror of the formatting (tab vs space I guess)

second thing, 3 large blocks of code that do the same thing, should have made a lambda or yet another method that takes a buffer ptr and a size